migration/transitioning.rst

```diff
@@ -0,0 +1,92 @@
+
+When the job is complete our schema will look like this:
+
+.. image:: ../images/erd/mt-after.png
+
+We call the tables considered so far *per-tenant* because querying them for our use case requires information for only one tenant per query. Their rows are distributed across the cluster according to the hashed values of their tenant ids.
+
+There are other types of tables to consider during a transition to Citus. Some are system-wide tables such as information about site administrators. We call them *global* tables and they do not participate in join queries with the per-tenant tables and may remain on the Citus coordinator node unmodified.
+
+Another kind of table are those which join with per-tenant tables but which aren't naturally specific to any one tenant. We call them *reference* tables. Two examples are shipping regions and product categories. We advise that you add a tenant id to these tables and duplicate the original rows, once for each tenant. This ensures that reference data is co-located with per-tenant data and quickly accessible to queries.
+
+Data Migration
+--------------
+
+Once the schema is updated and the per-tenant and reference tables are distributed across the cluster it's time to copy data from the original database into Citus. Most per-tenant tables can be copied directly from source tables. However line_items was denormalized with the addition of the  store_id column. We should copy data not from the original line_items table but from the results of a query on the source database:
+
+.. code:: sql
+
+  SELECT orders.store_id AS store_id, line_items.*
+    FROM line_items, orders
+   WHERE line_items.order_id = orders.order_id
+
+We call this *backfilling* the tenant id. To learn how to ingest datasets such as this into a Citus cluster, see :ref:`dml`.
+
+Query Migration
+---------------
+
+To execute queries efficiently for a specific tenant Citus needs to route them to the appropriate node and run them there. Thus every query must identify which tenant it involves. For simple select, update, and delete queries this means that the *where* clause must filter by tenant id.
+
+Suppose we want to get the details for an order. It used to suffice to filter by order_id. However once orders are distributed by store_id we must include that in the where filter as well.
+
+.. code:: sql
+
+  -- before
+  SELECT * FROM orders WHERE order_id = 123;
+
+  -- after
+  SELECT * FROM orders WHERE order_id = 123 AND store_id = 42;
+
+Likewise insert statements must always include a value for the tenant id column. Citus inspects that value for routing the insert command.
+
+When joining tables make sure they are all filtered by tenant id. For instance here is how to inspect how many awesome wool pants a given store has sold:
+
+.. code:: sql
+
+  SELECT sum(l.quantity)
```

Comment:
We discussed two options for modifying join queries in our discussion with Sai. The first option is the one you've documented, but the other option was:

- Filter one of the tables on store_id *AND* Join both tables on store_id as well.

Could we document this as well, with the example query?